client/src/components/Applications/AppForm/AppForm.jsx

```diff
@@ -150,6 +150,19 @@ const AdjustValuesWhenQueueChanges = ({ app }) => {
 };
 
 const AppInfo = ({ app }) => {
+  const categoryDict = useSelector((state) => state.apps.categoryDict);
+  const getAppCategory = (appId) => {
+    let category = null;
+    Object.entries(categoryDict).forEach(([cat, apps]) => {
+      if (apps.some((app) => app.appId === appId)) {
+        category = cat;
+      }
+    });
+    return category;
+  };
```

Comment:
const getAppCategory = (appId) => {
```suggestion
  for (const [cat, apps] of Object.entries(categoryDict)) {
      if (apps.some((app) => app.appId === appId)) {
        return cat;
      }
    }
    return null;
  };
```
   
A minor perf related question - the above code continues to look for category of an app, if it finds one. It gets the last matching category for an appid. Generally, an appid to category is 1:1. 
A suggestion to just return on first match. Feel free to use that or adjust to a better one. Also, feel free to let me know if this is accurate?